src/cascadia/TerminalApp/TerminalPaneContent.cpp

```diff
@@ -304,8 +314,15 @@ namespace winrt::TerminalApp::implementation
         RestartTerminalRequested.raise(*this, nullptr);
     }
 
-    void TerminalPaneContent::UpdateSettings(const TerminalSettingsCreateResult& settings,
-                                             const Profile& profile)
+    void TerminalPaneContent::UpdateSettings(const CascadiaSettings& /*settings*/)
+    {
+        // Do nothing. We'll later be updated manually by
+        // UpdateTerminalSettings, which we need for profile and
+        // focused/unfocused settings.
+    }

```

Comment:
Generally not a fan of do-nothing functions. Do we want `UpdateSettings()` to be on `SettingsPaneContent`? Then to update settings you have to check what kind of pane content it is then call the relevant function on it?

Reply:
Or maybe throw an exception so that the implementer knows to call `UpdateTerminalSettings` instead?

Reply:
Okay I added a helper class to make this a lot less painful.